### ISSUE TITLE
Enum Support breaks in Kogito 1.27.0 Need urgent fix

### ISSUE THREAD

**Issue by debu999** [kiegroup/kogito-runtimes#2436]
### Describe the bug

Enum in Java Classes which are part of BPMN does not work and breaks. Need a urgent fix. Seems to be a code change issue.

Breaking code = addEnumField
ordinal = Enum.valueOf((Class<Enum>) Class.forName(clazzName), field.name()).ordinal();
//1.27.0.FINAL
```
private void addEnumField(FieldInfo field, ProtoEnum pEnum) {
    AnnotationInstance annotation = field.annotation(ENUM_VALUE_ANNOTATION);
    Integer ordinal = null;
    boolean sortedWithAnnotation = false;
    if (annotation != null) {
        AnnotationValue number = annotation.value("number");
        if (number != null) {
            sortedWithAnnotation = true;
            ordinal = number.asInt();
        }
    }
    if (ordinal == null) {
        String clazzName = field.type().name().toString();
        try {
            ordinal = Enum.valueOf((Class<Enum>) Class.forName(clazzName), field.name()).ordinal();
        } catch (ClassNotFoundException e) {
            throw new RuntimeException("Failed to find enum class " + clazzName + " " + e.getMessage(), e);
        }
    }
    pEnum.addField(field.name(), ordinal, sortedWithAnnotation);
}
```
//1.26.0.Final
```
private void addEnumField(FieldInfo field, ProtoEnum pEnum) {
        AnnotationInstance annotation = field.annotation(ENUM_VALUE_ANNOTATION);
        Integer ordinal = null;
        boolean sortedWithAnnotation = false;
        if (annotation != null) {
            AnnotationValue number = annotation.value("number");
            if (number != null) {
                sortedWithAnnotation = true;
                ordinal = number.asInt();
            }
        }
        if (ordinal == null) {
            ordinal = pEnum.getFields()
                    .values()
                    .stream()
                    .mapToInt(Integer::intValue)
                    .max()
                    .orElse(-1) + 1;
        }
        pEnum.addField(field.name(), ordinal, sortedWithAnnotation);
    }
```

### Expected behavior

Enums should be working as earlier and support to the Enums should not have a issue in kogito.

### Actual behavior

```


Connected to the target VM, address: 'localhost:51860', transport: 'socket'
Press [h] for more options>
Tests paused
Press [r] to resume testing, [h] for more options>
Press [r] to resume testing, [o] Toggle test output, [h] for more options>
2022-09-04 18:05:48,487 INFO  [org.kie.kog.qua.rul.dep.RulesAssetsProcessor] (build-22) reflectiveEfestoRules()
2022-09-04 18:05:48,492 INFO  [org.kie.kog.qua.rul.dep.RulesAssetsProcessor] (build-22) toReturn {}
2022-09-04 18:05:48,494 INFO  [org.kie.kog.qua.pre.dep.PredictionsAssetsProcessor] (build-13) predictionSPIEvaluator()
2022-09-04 18:05:48,505 INFO  [org.kie.kog.qua.pre.dep.PredictionsAssetsProcessor] (build-28) reflectiveEfestoPredictions()
2022-09-04 18:05:48,512 INFO  [org.kie.efe.qua.dep.EfestoProcessor] (build-29) reflectiveEfestoStaticClasses
2022-09-04 18:05:48,573 INFO  [org.kie.kog.qua.pre.dep.PredictionsAssetsProcessor] (build-27) predictionSPIRuntime()
2022-09-04 18:05:48,809 INFO  [org.kie.kog.cod.api.uti.AddonsConfigDiscovery] (build-7) Performed addonsConfig discovery, found: AddonsConfig{usePersistence=true, useTracing=false, useMonitoring=false, usePrometheusMonitoring=false, useCloudEvents=false, useExplainability=false, useProcessSVG=false, useEventDrivenDecisions=false, useEventDrivenRules=false}
2022-09-04 18:05:48,970 INFO  [org.tes.doc.DockerClientProviderStrategy] (build-16) Loaded org.testcontainers.dockerclient.UnixSocketClientProviderStrategy from ~/.testcontainers.properties, will try it first
2022-09-04 18:05:49,181 WARN  [org.tes.doc.DockerClientProviderStrategy] (build-16) DOCKER_HOST unix:///var/run/docker.sock is not listening
2022-09-04 18:05:49,185 INFO  [org.tes.doc.DockerMachineClientProviderStrategy] (build-16) docker-machine executable was not found on PATH ([/usr/local/bin, /usr/bin, /bin, /usr/sbin, /sbin, /usr/local/sbin])
2022-09-04 18:05:49,187 ERROR [org.tes.doc.DockerClientProviderStrategy] (build-16) Could not find a valid Docker environment. Please check configuration. Attempted configurations were:
2022-09-04 18:05:49,187 ERROR [org.tes.doc.DockerClientProviderStrategy] (build-16) As no valid configuration was found, execution cannot continue
2022-09-04 18:05:49,240 INFO  [org.kie.kog.cod.cor.uti.ApplicationGeneratorDiscovery] (build-23) Generator discovery performed, found [processes, rules, decisions, predictions]
2022-09-04 18:05:49,243 INFO  [org.kie.kog.cod.cor.ApplicationGenerator] (build-23) Skipping generator 'decisions' because disabled
2022-09-04 18:05:49,243 INFO  [org.kie.kog.cod.cor.ApplicationGenerator] (build-23) Skipping generator 'predictions' because disabled
2022-09-04 18:05:49,995 INFO  [org.kie.kog.cod.rul.RuleCodegen] (build-23) No rule unit is present: generate KieRuntimeBuilder implementation.
2022-09-04 18:05:50,166 INFO  [org.kie.kog.qua.com.dep.KogitoAssetsProcessor] (build-22) reflectiveEfestoGeneratedClassBuildItem org.kie.kogito.quarkus.common.deployment.KogitoGeneratedSourcesBuildItem@41b05e10
2022-09-04 18:05:50,168 INFO  [org.kie.efe.qua.dep.EfestoProcessor] (build-23) reflectiveClassBuildItems {}
2022-09-04 18:05:50,344 INFO  [org.kie.kog.qua.pro.dep.KogitoDevServicesProcessor] (build-16) Dev Services for Kogito Data Index using image quay.io/kiegroup/kogito-data-index-ephemeral:1.27
2022-09-04 18:05:50,725 WARN  [org.kie.kog.qua.pro.dep.KogitoDevServicesProcessor] (build-16) Docker isn't working, unable to start Data Index image.
2022-09-04 18:06:51,756 INFO  [io.qua.dep.dev.IsolatedDevModeMain] (main) Attempting to start live reload endpoint to recover from previous Quarkus startup failure
2022-09-04 18:06:51,937 INFO  [org.jbo.threads] (main) JBoss Threads version 3.4.2.Final
2022-09-04 18:06:52,235 ERROR [io.qua.dep.dev.IsolatedDevModeMain] (main) Failed to start quarkus: java.lang.RuntimeException: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
	[error]: Build step org.kie.kogito.quarkus.processes.deployment.ProcessesAssetsProcessor#postGenerationProcessing threw an exception: java.lang.RuntimeException: Error while generating proto for model class org.doogle.enums.Test Failed to find enum class org.doogle.enums.DummyEnums org.doogle.enums.DummyEnums
	at org.kie.kogito.codegen.process.persistence.proto.AbstractProtoGenerator.generate(AbstractProtoGenerator.java:157)
	at org.kie.kogito.codegen.process.persistence.proto.AbstractProtoGenerator.protoOfDataClasses(AbstractProtoGenerator.java:52)
	at org.kie.kogito.codegen.process.persistence.PersistenceGenerator.generateProtoMarshaller(PersistenceGenerator.java:147)
	at org.kie.kogito.codegen.process.persistence.PersistenceGenerator.internalGenerate(PersistenceGenerator.java:125)
	at org.kie.kogito.codegen.core.AbstractGenerator.generate(AbstractGenerator.java:69)
	at org.kie.kogito.quarkus.processes.deployment.ProcessesAssetsProcessor.getGeneratedPersistenceFiles(ProcessesAssetsProcessor.java:298)
	at org.kie.kogito.quarkus.processes.deployment.ProcessesAssetsProcessor.generatePersistenceInfo(ProcessesAssetsProcessor.java:273)
	at org.kie.kogito.quarkus.processes.deployment.ProcessesAssetsProcessor.postGenerationProcessing(ProcessesAssetsProcessor.java:243)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:909)
	at io.quarkus.builder.BuildContext.run(BuildContext.java:281)
	at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2449)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1478)
	at java.base/java.lang.Thread.run(Thread.java:833)
	at org.jboss.threads.JBossThread.run(JBossThread.java:501)
Caused by: java.lang.RuntimeException: Failed to find enum class org.doogle.enums.DummyEnums org.doogle.enums.DummyEnums
	at org.kie.kogito.quarkus.processes.deployment.JandexProtoGenerator.addEnumField(JandexProtoGenerator.java:232)
	at org.kie.kogito.quarkus.processes.deployment.JandexProtoGenerator.lambda$enumFromClass$2(JandexProtoGenerator.java:207)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	at java.base/java.util.stream.SortedOps$RefSortingSink.end(SortedOps.java:395)
	at java.base/java.util.stream.Sink$ChainedReference.end(Sink.java:258)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:510)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
	at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:596)
	at org.kie.kogito.quarkus.processes.deployment.JandexProtoGenerator.lambda$enumFromClass$3(JandexProtoGenerator.java:207)
	at java.base/java.util.Optional.map(Optional.java:260)
	at org.kie.kogito.quarkus.processes.deployment.JandexProtoGenerator.enumFromClass(JandexProtoGenerator.java:202)
	at org.kie.kogito.quarkus.processes.deployment.JandexProtoGenerator.enumFromClass(JandexProtoGenerator.java:52)
	at org.kie.kogito.codegen.process.persistence.proto.AbstractProtoGenerator.internalGenerate(AbstractProtoGenerator.java:168)
	at org.kie.kogito.quarkus.processes.deployment.JandexProtoGenerator.messageFromClass(JandexProtoGenerator.java:148)
	at org.kie.kogito.quarkus.processes.deployment.JandexProtoGenerator.messageFromClass(JandexProtoGenerator.java:52)
	at org.kie.kogito.codegen.process.persistence.proto.AbstractProtoGenerator.internalGenerate(AbstractProtoGenerator.java:183)
	at org.kie.kogito.codegen.process.persistence.proto.AbstractProtoGenerator.generate(AbstractProtoGenerator.java:150)
	... 18 more
Caused by: java.lang.ClassNotFoundException: org.doogle.enums.DummyEnums
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:641)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:188)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:520)
	at io.quarkus.bootstrap.classloading.QuarkusClassLoader.loadClass(QuarkusClassLoader.java:490)
	at io.quarkus.bootstrap.classloading.QuarkusClassLoader.loadClass(QuarkusClassLoader.java:440)
	at java.base/java.lang.Class.forName0(Native Method)
	at java.base/java.lang.Class.forName(Class.java:375)
	at org.kie.kogito.quarkus.processes.deployment.JandexProtoGenerator.addEnumField(JandexProtoGenerator.java:230)
	... 38 more

	at io.quarkus.runner.bootstrap.AugmentActionImpl.runAugment(AugmentActionImpl.java:330)
	at io.quarkus.runner.bootstrap.AugmentActionImpl.createInitialRuntimeApplication(AugmentActionImpl.java:252)
	at io.quarkus.runner.bootstrap.AugmentActionImpl.createInitialRuntimeApplication(AugmentActionImpl.java:60)
	at io.quarkus.deployment.dev.IsolatedDevModeMain.firstStart(IsolatedDevModeMain.java:87)
	at io.quarkus.deployment.dev.IsolatedDevModeMain.accept(IsolatedDevModeMain.java:448)
	at io.quarkus.deployment.dev.IsolatedDevModeMain.accept(IsolatedDevModeMain.java:60)
	at io.quarkus.bootstrap.app.CuratedApplication.runInCl(CuratedApplication.java:148)
	at io.quarkus.bootstrap.app.CuratedApplication.runInAugmentClassLoader(CuratedApplication.java:103)
	at io.quarkus.deployment.dev.DevModeMain.start(DevModeMain.java:131)
	at io.quarkus.deployment.dev.DevModeMain.main(DevModeMain.java:62)
Caused by: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
	[error]: Build step org.kie.kogito.quarkus.processes.deployment.ProcessesAssetsProcessor#postGenerationProcessing threw an exception: java.lang.RuntimeException: Error while generating proto for model class org.doogle.enums.Test Failed to find enum class org.doogle.enums.DummyEnums org.doogle.enums.DummyEnums
	at org.kie.kogito.codegen.process.persistence.proto.AbstractProtoGenerator.generate(AbstractProtoGenerator.java:157)
	at org.kie.kogito.codegen.process.persistence.proto.AbstractProtoGenerator.protoOfDataClasses(AbstractProtoGenerator.java:52)
	at org.kie.kogito.codegen.process.persistence.PersistenceGenerator.generateProtoMarshaller(PersistenceGenerator.java:147)
	at org.kie.kogito.codegen.process.persistence.PersistenceGenerator.internalGenerate(PersistenceGenerator.java:125)
	at org.kie.kogito.codegen.core.AbstractGenerator.generate(AbstractGenerator.java:69)
	at org.kie.kogito.quarkus.processes.deployment.ProcessesAssetsProcessor.getGeneratedPersistenceFiles(ProcessesAssetsProcessor.java:298)
	at org.kie.kogito.quarkus.processes.deployment.ProcessesAssetsProcessor.generatePersistenceInfo(ProcessesAssetsProcessor.java:273)
	at org.kie.kogito.quarkus.processes.deployment.ProcessesAssetsProcessor.postGenerationProcessing(ProcessesAssetsProcessor.java:243)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:909)
	at io.quarkus.builder.BuildContext.run(BuildContext.java:281)
	at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2449)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1478)
	at java.base/java.lang.Thread.run(Thread.java:833)
	at org.jboss.threads.JBossThread.run(JBossThread.java:501)
Caused by: java.lang.RuntimeException: Failed to find enum class org.doogle.enums.DummyEnums org.doogle.enums.DummyEnums
	at org.kie.kogito.quarkus.processes.deployment.JandexProtoGenerator.addEnumField(JandexProtoGenerator.java:232)
	at org.kie.kogito.quarkus.processes.deployment.JandexProtoGenerator.lambda$enumFromClass$2(JandexProtoGenerator.java:207)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	at java.base/java.util.stream.SortedOps$RefSortingSink.end(SortedOps.java:395)
	at java.base/java.util.stream.Sink$ChainedReference.end(Sink.java:258)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:510)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
	at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:596)
	at org.kie.kogito.quarkus.processes.deployment.JandexProtoGenerator.lambda$enumFromClass$3(JandexProtoGenerator.java:207)
	at java.base/java.util.Optional.map(Optional.java:260)
	at org.kie.kogito.quarkus.processes.deployment.JandexProtoGenerator.enumFromClass(JandexProtoGenerator.java:202)
	at org.kie.kogito.quarkus.processes.deployment.JandexProtoGenerator.enumFromClass(JandexProtoGenerator.java:52)
	at org.kie.kogito.codegen.process.persistence.proto.AbstractProtoGenerator.internalGenerate(AbstractProtoGenerator.java:168)
	at org.kie.kogito.quarkus.processes.deployment.JandexProtoGenerator.messageFromClass(JandexProtoGenerator.java:148)
	at org.kie.kogito.quarkus.processes.deployment.JandexProtoGenerator.messageFromClass(JandexProtoGenerator.java:52)
	at org.kie.kogito.codegen.process.persistence.proto.AbstractProtoGenerator.internalGenerate(AbstractProtoGenerator.java:183)
	at org.kie.kogito.codegen.process.persistence.proto.AbstractProtoGenerator.generate(AbstractProtoGenerator.java:150)
	... 18 more
Caused by: java.lang.ClassNotFoundException: org.doogle.enums.DummyEnums
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:641)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:188)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:520)
	at io.quarkus.bootstrap.classloading.QuarkusClassLoader.loadClass(QuarkusClassLoader.java:490)
	at io.quarkus.bootstrap.classloading.QuarkusClassLoader.loadClass(QuarkusClassLoader.java:440)
	at java.base/java.lang.Class.forName0(Native Method)
	at java.base/java.lang.Class.forName(Class.java:375)
	at org.kie.kogito.quarkus.processes.deployment.JandexProtoGenerator.addEnumField(JandexProtoGenerator.java:230)
	... 38 more

	at io.quarkus.builder.Execution.run(Execution.java:123)
	at io.quarkus.builder.BuildExecutionBuilder.execute(BuildExecutionBuilder.java:79)
	at io.quarkus.deployment.QuarkusAugmentor.run(QuarkusAugmentor.java:158)
	at io.quarkus.runner.bootstrap.AugmentActionImpl.runAugment(AugmentActionImpl.java:328)
	... 9 more
Caused by: java.lang.RuntimeException: Error while generating proto for model class org.doogle.enums.Test Failed to find enum class org.doogle.enums.DummyEnums org.doogle.enums.DummyEnums
	at org.kie.kogito.codegen.process.persistence.proto.AbstractProtoGenerator.generate(AbstractProtoGenerator.java:157)
	at org.kie.kogito.codegen.process.persistence.proto.AbstractProtoGenerator.protoOfDataClasses(AbstractProtoGenerator.java:52)
	at org.kie.kogito.codegen.process.persistence.PersistenceGenerator.generateProtoMarshaller(PersistenceGenerator.java:147)
	at org.kie.kogito.codegen.process.persistence.PersistenceGenerator.internalGenerate(PersistenceGenerator.java:125)
	at org.kie.kogito.codegen.core.AbstractGenerator.generate(AbstractGenerator.java:69)
	at org.kie.kogito.quarkus.processes.deployment.ProcessesAssetsProcessor.getGeneratedPersistenceFiles(ProcessesAssetsProcessor.java:298)
	at org.kie.kogito.quarkus.processes.deployment.ProcessesAssetsProcessor.generatePersistenceInfo(ProcessesAssetsProcessor.java:273)
	at org.kie.kogito.quarkus.processes.deployment.ProcessesAssetsProcessor.postGenerationProcessing(ProcessesAssetsProcessor.java:243)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:909)
	at io.quarkus.builder.BuildContext.run(BuildContext.java:281)
	at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2449)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1478)
	at java.base/java.lang.Thread.run(Thread.java:833)
	at org.jboss.threads.JBossThread.run(JBossThread.java:501)
Caused by: java.lang.RuntimeException: Failed to find enum class org.doogle.enums.DummyEnums org.doogle.enums.DummyEnums
	at org.kie.kogito.quarkus.processes.deployment.JandexProtoGenerator.addEnumField(JandexProtoGenerator.java:232)
	at org.kie.kogito.quarkus.processes.deployment.JandexProtoGenerator.lambda$enumFromClass$2(JandexProtoGenerator.java:207)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	at java.base/java.util.stream.SortedOps$RefSortingSink.end(SortedOps.java:395)
	at java.base/java.util.stream.Sink$ChainedReference.end(Sink.java:258)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:510)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
	at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:596)
	at org.kie.kogito.quarkus.processes.deployment.JandexProtoGenerator.lambda$enumFromClass$3(JandexProtoGenerator.java:207)
	at java.base/java.util.Optional.map(Optional.java:260)
	at org.kie.kogito.quarkus.processes.deployment.JandexProtoGenerator.enumFromClass(JandexProtoGenerator.java:202)
	at org.kie.kogito.quarkus.processes.deployment.JandexProtoGenerator.enumFromClass(JandexProtoGenerator.java:52)
	at org.kie.kogito.codegen.process.persistence.proto.AbstractProtoGenerator.internalGenerate(AbstractProtoGenerator.java:168)
	at org.kie.kogito.quarkus.processes.deployment.JandexProtoGenerator.messageFromClass(JandexProtoGenerator.java:148)
	at org.kie.kogito.quarkus.processes.deployment.JandexProtoGenerator.messageFromClass(JandexProtoGenerator.java:52)
	at org.kie.kogito.codegen.process.persistence.proto.AbstractProtoGenerator.internalGenerate(AbstractProtoGenerator.java:183)
	at org.kie.kogito.codegen.process.persistence.proto.AbstractProtoGenerator.generate(AbstractProtoGenerator.java:150)
	... 18 more
Caused by: java.lang.ClassNotFoundException: org.doogle.enums.DummyEnums
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:641)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:188)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:520)
	at io.quarkus.bootstrap.classloading.QuarkusClassLoader.loadClass(QuarkusClassLoader.java:490)
	at io.quarkus.bootstrap.classloading.QuarkusClassLoader.loadClass(QuarkusClassLoader.java:440)
	at java.base/java.lang.Class.forName0(Native Method)
	at java.base/java.lang.Class.forName(Class.java:375)
	at org.kie.kogito.quarkus.processes.deployment.JandexProtoGenerator.addEnumField(JandexProtoGenerator.java:230)
	... 38 more

Disconnected from the target VM, address: 'localhost:51860', transport: 'socket'
```

### How to Reproduce?

Any sample kogito program app with a java class containing Enum will break in 1.27.0.Final.

Sample java class
```
package org.doogle.enums;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class Test {
    public DummyEnums dummyEnums;
}

```

sample enum
```
package org.doogle.enums;

public enum DummyEnums {
    TEST,
    TEST1
}

```

since DummyEnums is in Test class and test class is in bpmn enum.bpmn2

### Output of `uname -a` or `ver`

Darwin Debabratas-MacBook-Pro.local 21.5.0 Darwin Kernel Version 21.5.0: Tue Apr 26 21:08:22 PDT 2022; root:xnu-8020.121.3~4/RELEASE_X86_64 x86_64

### Output of `java -version`

openjdk version "17.0.2" 2022-01-18 OpenJDK Runtime Environment GraalVM CE 22.0.0.2 (build 17.0.2+8-jvmci-22.0-b05) OpenJDK 64-Bit Server VM GraalVM CE 22.0.0.2 (build 17.0.2+8-jvmci-22.0-b05, mixed mode, sharing)

### GraalVM version (if different from Java)

GraalVM CE 22.0.0.2 (build 17.0.2+8-jvmci-22.0-b05)

### Kogito version or git rev (or at least Quarkus version if you are using Kogito via Quarkus platform BOM)

1.27.0.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Maven home: /usr/local/Cellar/maven/3.8.5/libexec Java version: 18.0.1.1, vendor: Homebrew, runtime: /usr/local/Cellar/openjdk/18.0.1.1/libexec/openjdk.jdk/Contents/Home Default locale: en_IN, platform encoding: UTF-8 OS name: "mac os x", version: "12.4", arch: "x86_64", family: "mac"

### Additional information

code change have caused issue in Enum suppport in kogito 1.27.0.Final

**Comment by debu999** [comment]
@cristianonicolai @fjtirado is it a change to fix something which broke the base calls. BTW... is there a way to  load the classes forcefully in quarkus. Guess that might solve the issue if classloader can be updated for the class in concern.

**Comment by cristianonicolai** [comment]
Fix merged, and should be available on version 1.28.0
